source/common/router/router.h

```diff
@@ -206,6 +206,7 @@ class Filter : Logger::Loggable<Logger::Id::router>, public Http::StreamDecoderF
   const VirtualCluster* request_vcluster_;
   Event::TimerPtr response_timeout_;
   FilterUtility::TimeoutData timeout_;
+  Http::Code timeout_response_code_;
```

Comment:
just initialize to Http::Code::GatewayTimeout here, and avoid else case in other file.